### PR TITLE
WIP: EventSub

### PR DESCRIPTION
**Pull request by ghostzero**
I added the new event sub implementation.

I have also implemented an [event handler](https://github.com/ghostzero/Laravel-Twitch/tree/webhooks-2.0#defining-websub-event-handlers) to simplify the handling of EventSub (including Secret + Signature check). So you only have to extend the `EventSubController` and register your controller in the routes.

Feel free to share your feedback 😃

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -281,84 +370,125 @@ public function updateUserExtensions(?string $method = NULL, ?string $parameter
...
+), array $body = array (
+))
 ```
 
 ### Videos
 
 ```php
-public function getVideos(array $parameters = [], ?Paginator $paginator = NULL)
+public function getVideos(array $parameters = array (
+), ?Paginator $paginator = NULL)
```
Comment (ghostzero) [inline README.md:441]: Hmm, the docs generator breaks the format

File: src/Enums/EventSubType.php [hunk 2]
```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace romanzipp\Twitch\Enums;
+
+class EventSubType
```
Comment (ghostzero) [inline src/Enums/EventSubType.php:5]: Check all consts after public release, this are still the beta contants.

**Review by ghostzero — COMMENTED** [review]
I just had another look at the source code, because I haven't worked on it for a while. I noticed some small things. Especially the one with the signature check and handling.

**Comment by ghostzero** [comment]
Added tests for api, constrollers, and signature validation ✔️

File: README.md [hunk 3]
```diff
@@ -266,6 +330,14 @@ public function getDropsEntitlements(array $parameters = [])
 public function redeemEntitlementsCode(array $parameters = [])
 ```
 
+### EventSub
+
+```php
+public function subscribeEventSub(array $parameters = [], array $body = [])
```
Comment (ghostzero) [inline README.md:336]: I don't know yet if we want to leave the payload in there, officially you don't need it.

File: README.md [hunk 4]
```diff
@@ -218,6 +218,70 @@ $result->insertUsers($twitch, 'from_id', 'from_user');
...
+);
+```
+
+### Subscribe EventSub Events
+
+```php
+$twitch = new romanzipp\Twitch\Twitch;
+
+
+$twitch->subscribeEventSub([
```
Comment (ghostzero) [inline README.md:271]: first attribute is missing

```
...Sub([], [...
```

**Comment by romanzipp** [comment]
@ghostzero could you rebase the PR?

**Comment by ghostzero** [comment]
Done 👍 This was my first rebase 🔥 😄

**Pull request merged**